aoc-client/src/lib.rs

```diff
@@ -421,6 +421,79 @@ impl AocClient {
         Ok(())
     }
 
+    fn get_event_stats_html(&self) -> AocResult<String> {
+        debug!("🦌 Fetching event stats for {}", self.year);
+        let url = format!("https://adventofcode.com/{}/stats", self.year);
+
+        let response = http_client(&self.session_cookie, "text/html")?
+            .get(url)
+            .send()?;
+        let contents = response.error_for_status()?.text()?;
+
+        let main = Regex::new(r"(?i)(?s)<main>(?P<main>.*)</main>")
+            .unwrap()
+            .captures(&contents)
+            .ok_or(AocError::AocResponseError)?
+            .name("main")
+            .unwrap()
+            .as_str()
+            .to_string();
+
+        Ok(main)
+    }
+
+    pub fn show_event_stats(&self) -> AocResult<()> {
+        let stats_html = self.get_event_stats_html()?;
+
+        let stats_text = from_read_with_decorator(
+            stats_html.as_bytes(),
+            self.output_width,
+            TrivialDecorator::new(),
+        );
+
+        // number of completions loosely associated with each star
+        let star_val = stats_text
+            .split_whitespace()
+            .find(|chunk| chunk.parse::<u64>().is_ok())
+            .unwrap()
+            .parse::<u64>()
+            .unwrap();
+
+        // add color to appropriate text in explanatory sentence
+        let stats_colored = stats_text
+            .replace("Gold", &"Gold".color(GOLD).to_string())
+            .replace("silver", &"silver".color(SILVER).to_string())
+            .replacen("*", &"*".color(SILVER).to_string(), 2)
+            .replacen("*", &"*".color(GOLD).to_string(), 1);
+
+        println!("");
+        for line in stats_colored.lines().take_while(|line| !line.is_empty()) {
+            println!("{}", line);
+        }
+
+        for line in stats_colored.lines().skip_while(|line| !line.is_empty()) {
+            if line.is_empty() {
+                continue;
+            }
+            let split: Vec<&str> = line.split_whitespace().collect();
+            let gold_comps = split[1].parse::<usize>().unwrap();
+            let n_stars = split[3].len();
```

Comment:
This also assumes the page format won't change. Maybe a regex would be safer (and return error if it doesn't match).

Reply:
Likewise here, I will start working on switching this functionality over to a regex. Thanks so much for the feedback!